packages/ember-data/lib/system/relationships/state/has_many.js

```diff
@@ -16,6 +16,25 @@ ManyRelationship.prototype = Ember.create(Relationship.prototype);
 ManyRelationship.prototype.constructor = ManyRelationship;
 ManyRelationship.prototype._super$constructor = Relationship;
 
+ManyRelationship.prototype.setup = function(data) {
+  var key   = this.key,
+      value = data[key];
+
+  if (data.links && data.links[key]) {
+    this.updateLink(data.links[key]);
+  }
+
+  if (this.buildURL) {
```

Comment:
Basically instead of the option, if no value and no link, we should set a flag that is 'generateLink' if there is a buildURLForHasMany in the adapter